### PR TITLE
Add GitHub workflows Merge with Main

### PR DESCRIPTION
**Pull request by CassandraGross1**
Need ci-cd.yml reviewed and merged with main branch after review.

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      uses: aws-actions/eks-update-kubeconfig@v1  # Set up kubeconfig to allow kubectl to communicate with the Kubernetes cluster
+      with:
+        region: ${{ env.AWS_REGION }}
+        cluster-name: ${{ env.EKS_CLUSTER_NAME }}
+        role-arn: ${{ secrets.AWS_ROLE_ARN }}  # AWS IAM role for accessing the EKS cluster, stored securely in GitHub secrets DO WE NEED THIS? 
+
+    - name: Deploy Infrastructure to Kubernetes
+      run: |
+        # Deploy the namespace to logically separate resources within the Kubernetes cluster
+        kubectl apply -f k8s/namespace.yml -n ${{ env.K8S_NAMESPACE }}
```
Comment (zimmermantr) [inline .github/workflows/ci-cd.yml:43]: I don't think you need to specify the namespace when applying the namespace.

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        # Deploy other infrastructure-related resources (e.g., storage classes, resource quotas)
+        kubectl apply -f k8s/ -n ${{ env.K8S_NAMESPACE }} --exclude k8s/namespace.yml
```
Comment (zimmermantr) [inline .github/workflows/ci-cd.yml:45]: I don't think you need to exclude the namespace file, but it should still work.

**Review by zimmermantr — APPROVED** [review]

File: .github/workflows/ci-cd.yml (hunk 1 above)
Comment (CassandraGross1) [reply .github/workflows/ci-cd.yml:43]: LOL.  True.

**Comment by CassandraGross1** [comment]
Merging corrected infrastructure ci-cd.yml with Tristan's approval

**Pull request merged**